Apps/Medication/src/Delegates/SalesforceDelegate.cs

```diff
@@ -98,71 +98,20 @@ public async Task<RequestResult<IList<MedicationRequest>>> GetMedicationRequests
-                            this.logger.LogTrace("Response payload: {Payload}", payload);
-                            ResponseWrapper? replyWrapper = JsonSerializer.Deserialize<ResponseWrapper>(payload);
-                            if (replyWrapper != null)
-                            {
-                                retVal.ResultStatus = ResultType.Success;
-                                retVal.ResourcePayload = this.autoMapper.Map<IEnumerable<SpecialAuthorityRequest>, IList<MedicationRequest>>(replyWrapper.Items);
-                                retVal.TotalResultCount = replyWrapper.Items.Count;
-                                retVal.PageSize = replyWrapper.Items.Count;
-                                retVal.PageIndex = 0;
-                            }
-                            else
-                            {
-                                retVal.ResultError = new RequestResultError
-                                {
-                                    ResultMessage = "Error with JSON data",
-                                    ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Sf),
-                                };
-                            }
-
-                            break;
-                        case HttpStatusCode.NoContent: // No Medication Requests exits for this user
-                            retVal.ResultStatus = ResultType.Success;
-                            retVal.ResourcePayload = new List<MedicationRequest>();
-                            retVal.TotalResultCount = 0;
-                            retVal.PageSize = 0;
-                            break;
-                        case HttpStatusCode.Forbidden:
-                            retVal.ResultError = new RequestResultError
-                            {
-                                ResultMessage = $"DID Claim is missing or can not resolve PHN, HTTP Error {response.StatusCode}",
-                                ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Sf),
-                            };
-                            break;
-                        default:
-                            retVal.ResultError = new RequestResultError
-                            {
-                                ResultMessage = $"Unable to connect to Medication Requests endpoint, HTTP Error {response.StatusCode}",
-                                ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Sf),
-                            };
-                            this.logger.LogError("Unable to connect to endpoint {Endpoint}, HTTP Error {StatusCode}\n{Payload}", endpoint, response.StatusCode, payload);
-                            break;
-                    }
+                    ResponseWrapper? replyWrapper = await this.specialAuthorityApi.GetSpecialAuthorityRequests(phn, accessToken).ConfigureAwait(true);
```

Comment:
Line 136 check for NoContent is removed.  Are you sure NoContent will not result in an ApiException.  Have you tried confirming this in a unit test? Have you tried mocking the response to return No Content?

var apiMockedResponse = 
        setupApiRequest.ReturnsAsync(new HttpResponseMessage() 
        { 
            StatusCode = HttpStatusCode.NoContent, 
            Content = null, 
        });

Reply:
I tested the code against a user without any Special Authority requests and the wrapper object was still returned.  I think that No Content has spread in our code base when in most places it is not used.